### ISSUE TITLE
Gallery multiply breakpoints options configuration doesn't work properly because of mediaCheck triggering exit media query function on page load

### ISSUE THREAD

**Issue by Januszpl** [magento/magento2#29933]
### Preconditions (*)
1. Magento 2.3 / 2.4-develop
2. Multiply breakpoints configured with custom options in `[your theme]/etc/view.xml`

### Steps to reproduce (*)
1. Define more than one breakpoints with custom options

```xml
    <vars module="Magento_Catalog">
        <var name="breakpoints">
            <var name="desktop">
                <var name="conditions">
                    <var name="min-width">1024px</var>
                </var>
                <var name="options">
                    <var name="options">
                        <var name="navdir">vertical</var>
                    </var>
                </var>
            </var>
            <var name="mobile">
                <var name="conditions">
                    <var name="max-width">767px</var>
                </var>
                <var name="options">
                    <var name="options">
                        <var name="nav">dots</var>
                    </var>
                </var>
            </var>
        </var>
    </vars>
```

2. Set mobile screen size
3. Load page

### Expected result (*)
Gallery should have "dots" navigation.

### Actual result (*)
Gallery have "thumbs" navigation.

**Severity**: S2 - S4

**Comment by Januszpl** [comment]
I will prepare PR for that.

**Comment by magento-engcom-team** [comment]
@engcom-Delta Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [ ] Add `"Component: "` label(s) to this ticket based on verification result. If uncertain, you may follow the best guess
- [ ] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Delta 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-37677` were created

**Issue Available**: @engcom-Delta, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @Januszpl. Thank you for your report. 
The issue has been fixed in magento/magento2#29934 by @Januszpl in 2.4-develop branch
Related commit(s):
 - [faf6dface4d7ac453c181fdad50a51e70e8e673c](https://github.com/magento/magento2/commit/faf6dface4d7ac453c181fdad50a51e70e8e673c)
 - [9e53105666f404b557165bf096aaf95c31388b27](https://github.com/magento/magento2/commit/9e53105666f404b557165bf096aaf95c31388b27)
 - [a46a395a1a6e3f40f90fb040874f518083ba9627](https://github.com/magento/magento2/commit/a46a395a1a6e3f40f90fb040874f518083ba9627)
 - [c9b7a0266aec9b79c42a16b6034d81fe9bdb7cd5](https://github.com/magento/magento2/commit/c9b7a0266aec9b79c42a16b6034d81fe9bdb7cd5)

The fix will be available with the upcoming 2.4.2 release.
